runtime/src/main/scala/akka/grpc/internal/NettyClientUtils.scala

```diff
@@ -31,12 +31,15 @@ object NettyClientUtils {
         .forAddress(settings.host, settings.port)
         .flowControlWindow(NettyChannelBuilder.DEFAULT_FLOW_CONTROL_WINDOW)
 
-    builder = settings.trustedCaCertificate.map(c => GrpcSslContexts.forClient.trustManager(loadCert(c)).build)
-      .map(ctx => builder.negotiationType(NegotiationType.TLS).sslContext(ctx))
-      .getOrElse(builder.negotiationType(NegotiationType.PLAINTEXT))
-
-    builder = settings.overrideAuthority.map(builder.overrideAuthority(_)).getOrElse(builder)
-
+    if (!settings.useTls)
+      builder = builder.usePlaintext()
+    else {
+      builder = settings.trustedCaCertificate.map(c => GrpcSslContexts.forClient.trustManager(loadCert(c)).build)
+        .map(ctx => builder.negotiationType(NegotiationType.TLS).sslContext(ctx))
+        .getOrElse(builder.negotiationType(NegotiationType.PLAINTEXT))
```

Comment:
I guess we want to throw an error instead of falling back to PLAINTEXT when `useTls` is enabled but `trustedCaCertificate` is nevertheless `None`?

Reply:
I didn't really change anything in this branch, just moved the original code down here, but maybe that is true.